python/phase-1.md

```diff
@@ -316,16 +442,18 @@ Notes:
   * When we get to workflows, we will do similar with `@workflow.run` vs `@workflow.signal` vs `@workflow.query`, etc
     and `workflow.info()`
 * Cancellation is task cancellation
-  * Do we need to differentiate worker shutdown from server-side cancellation?
-  * So we need an explicit future they can wait on for activity cancellation?
+  * We need to research whether, in Python 3.7, we can interrupt an async task w/ our own error reliably
+* Heartbeat is a synchronous call
+  * Should we throw a cancellation error if we know the activity is cancelled when they try to heartbeat?
+    * is this something core should be returning from `record_heartbeat` instead of lang?
```

Comment:
Yes and maybe also yes. We didn't return errors from heartbeating in core for reasons that I never thought were super great so we can revisit that decision for sure.

That said, for now, it's simple to throw if you already know you're cancelled